src/main/java/nl/hsac/fitnesse/fixture/util/selenium/SeleniumHelper.java

```diff
@@ -1057,6 +1063,44 @@ public String takeScreenshot(String baseName) {
         return result;
     }
 
+    /**
+     * Takes screenshot of a specific element on the page (as .png).
+     * @param baseName name for file created (without extension),
+     *                 if a file already exists with the supplied name an
+     *                 '_index' will be added.
+     * @param element the webelement to crop the image to.
+     * @return absolute path of file created.
+     */
+    public String takeElementScreenshot(String baseName, WebElement element) {
+        String result = null;
+
+        WebDriver d = driver();
+
+        if (!(d instanceof TakesScreenshot)) {
+            d = new Augmenter().augment(d);
+        }
+        if (d instanceof TakesScreenshot) {
+            TakesScreenshot ts = (TakesScreenshot) d;
+            File screenshot = ts.getScreenshotAs(OutputType.FILE);
```

Comment:
This first part seems to share logic with the existing screenshot functionality. Can you see whether that can be extracted to reduce code/functionality duplication?

Reply:
I extracted the augmenting and casting to a separate method that returns a TakesScreenshot instance. Let me know if that helps.